src/Event/RequestSubscriber.php

```diff
@@ -62,10 +69,31 @@ protected function getMethodForProvider($providerId)
   public function onRequest(GetResponseEvent $event, $eventName, EventDispatcherInterface $dispatcher)
   {
     $request = $event->getRequest();
-    $modifiers = $this->getModifiers();
+    $uri = $request->getPathInfo();
+    [, $uri] = explode('/', $uri);
+
+    // Eliminate the need to get and iterate through all the modifiers as it is
+    // a very resource intensive process.
+    $provider = Provider::load(self::GROUP_PURL_PROVIDER);
```

Comment:
We can not just call the GROUP_PURL_PROVIDER as we will soon have multiple providers for domain in addition to path prefix, also purl should not be dependent on group_purl.